

# <font color="#ff4500" style="font-size: 30px;">Código Python Todo Dia</font>


<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 108</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> Nos exercícios anteriores utilizamos compreensão de listas para construir listas de uma forma rápida e 'pythônica' utilizando uma notação simples. Os dicionários também suportam a mesma operação, possibilitando uma construção simples e compacta combinando loops e estruturas condicionais. Nestes termos, podemos aplicar esse conceito para criar DataFrames com pandas, uma aplicação muito comum em Data Science.

## Sendo assim, utilize compreensão de dicionários para construir os dicionários abaixo.
--- 
## a) Crie um dicionário em que suas chaves serão os números 1, 4, 5, 6, 7, 9 (que podem ser armazenados em uma lista) e seus valores correspondem ao quadrado destes números.


    
```python
{1: 1, 4: 16, 5: 25, 6: 36, 7: 49, 9: 81}
   
```
---
    
## b) Crie um dicionário em que suas chaves correspondem a números inteiros entre [1,10] e cada valor associado é o número ao quadrado.
    
```python
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100}
```
---
## c) Crie um dicionário em que suas chaves correspondem a números de 10 a 1 e seus respectivos valores denotam o cubo de cada chave.
    
```python
{10: 1000, 9: 729, 8: 512, 7: 343, 6: 216, 5: 125, 4: 64, 3: 27, 2: 8, 1: 1}
```
</b></i>     
</p>

---

In [6]:
# A)
lista = [1, 4, 5, 6, 7, 9]
dict_ = {x:x**2 for x in lista}
print(dict_)

# B)
dict_ = {x:x**2 for x in range(1,11)}
print(dict_)

# C)
dict_ = {x:x**3 for x in range(10,0,-1)}
print(dict_)

{1: 1, 4: 16, 5: 25, 6: 36, 7: 49, 9: 81}
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100}
{10: 1000, 9: 729, 8: 512, 7: 343, 6: 216, 5: 125, 4: 64, 3: 27, 2: 8, 1: 1}


<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 109</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> A partir da lista

    
```python
linguagens = ['python', 'javascript', 'java', 'php', 'c', 'julia']
   
```
---
    
## Use Dict Comprehension para construir o dicionário abaixo:   
    
```python
{'python': 6, 'javascript': 10, 'java': 4, 'php': 3, 'c': 1, 'julia': 5}
```
---
## Dica: observe que o valor associado a cada chave corresponde ao tamanho de cada string..
    
</b></i>     
</p>

---

In [7]:
linguagens = ['python', 'javascript', 'java', 'php', 'c', 'julia']

dict_ = {x:len(x) for x in linguagens}
dict_

{'python': 6, 'javascript': 10, 'java': 4, 'php': 3, 'c': 1, 'julia': 5}

<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 110</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> A partir da lista

    
```python
frutas = ['  Banana  ', ' Laranja   ', '  Maçã', ' Melão ']
   
```
---
    
## Use Dict Comprehension para construir o dicionário abaixo:   
    
```python
{'1': 'Banana', '2': 'Laranja', '3': 'Maçã', '4': 'Melão'}
```
---
    
</b></i>     
</p>

---

In [10]:
frutas = ['  Banana  ', ' Laranja   ', '  Maçã', ' Melão ']

dict_ = {str(indice):fruta.strip() for indice, fruta in enumerate(frutas, start=1)}
dict_

{'1': 'Banana', '2': 'Laranja', '3': 'Maçã', '4': 'Melão'}

<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 111</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> Utilize compreensão de dicionários para construir o dicionário abaixo.

    
```python
{'chave1': 1, 'chave2': 4, 'chave3': 9, 'chave4': 16, 'chave5': 25, 'chave6': 36, 'chave7': 49, 'chave8': 64, 'chave9': 81}
   
```
---
    
</b></i>     
</p>

---

In [2]:
dict_ = {f"chave{i}":i**2 for i in range(1,10)}
print(dict_)

{'chave1': 1, 'chave2': 4, 'chave3': 9, 'chave4': 16, 'chave5': 25, 'chave6': 36, 'chave7': 49, 'chave8': 64, 'chave9': 81}


<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 112</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> Utilize compreensão de dicionários para construir o dicionário abaixo.

    
```python
{'A': 1, 'B': 2, 'C': 6, 'D': 24, 'E': 120}
   
```

Dica: observe que o valor associado a cada chave corresponde ao fatorial de cada número no intervalo [1,5].
    
---
    
</b></i>     
</p>

---

In [21]:
import math as mp

lista = ['A','B','C','D','E']
dict_ = {element:mp.factorial(num) for num, element in enumerate(lista, start=1) }
print(dict_)

# outra resolução 

from math import factorial
dicionario = {chr(64+i):factorial(i)  for i in range(1,6)}
dicionario

{'A': 1, 'B': 2, 'C': 6, 'D': 24, 'E': 120}


{'A': 1, 'B': 2, 'C': 6, 'D': 24, 'E': 120}

<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 113</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> A partir das listas

    
```python
empresa = ['ABEV3', 'AZUL4', 'BTOW3', 'RENT3', 'JBSS3']
precos = [12.85, 22.55, 115.08, 49.03, 24.41] 
   
```

Utilize compreensão de dicionários e obtenha:

    
```python
{'ABEV3': 12.85, 'AZUL4': 22.55, 'BTOW3': 115.08, 'RENT3': 49.03, 'JBSS3': 24.41}
```
   
    
---
    
</b></i>     
</p>

---

In [14]:
empresa = ['ABEV3', 'AZUL4', 'BTOW3', 'RENT3', 'JBSS3']
precos = [12.85, 22.55, 115.08, 49.03, 24.41] 

dict_ = {E: P for E, P in zip(empresa, precos)}
print(dict_)

# Outra forma de resolver
dict_ = {empresa[i]:precos[i] for i in range(len(empresa))}
print(dict_)

{'ABEV3': 12.85, 'AZUL4': 22.55, 'BTOW3': 115.08, 'RENT3': 49.03, 'JBSS3': 24.41}
{'ABEV3': 12.85, 'AZUL4': 22.55, 'BTOW3': 115.08, 'RENT3': 49.03, 'JBSS3': 24.41}


<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 114</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> Do dicionário abaixo:

    
```python
dados = {12.85: 'ABEV3', 22.55: 'AZUL4', 115.08: 'BTOW3', 49.03: 'RENT3', 24.41: 'JBSS3'}
   
```

Utilize compreensão de dicionários e obtenha:

    
```python
{'ABEV3': 12.85, 'AZUL4': 22.55, 'BTOW3': 115.08, 'RENT3': 49.03, 'JBSS3': 24.41}
```
   
    
---
    
</b></i>     
</p>

---

In [19]:
dados = {12.85: 'ABEV3', 22.55: 'AZUL4', 115.08: 'BTOW3', 49.03: 'RENT3', 24.41: 'JBSS3'}

dict_ = {valor:chave for chave, valor in dados.items()}
print(dict_)

{'ABEV3': 12.85, 'AZUL4': 22.55, 'BTOW3': 115.08, 'RENT3': 49.03, 'JBSS3': 24.41}


<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 115</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> Use compreensão de dicionários e obtenha os dicionários a seguir:

    
```python
a)



{2: 1.0, 4: 2.0, 6: 2.585, 8: 3.0, 10: 3.322, 12: 3.585, 14: 3.807, 16: 4.0, 18: 4.17, 20: 4.322}
   
```

### Dica: este dicionário calcula o logaritmo de base 2 para todos os números pares no intervalo [1,20].

    
```python
b)



{'C': 3, 'E': 5, 'F': 6, 'I': 9, 'J': 10, 'L': 12, 'O': 15, 'R': 18, 'T': 20}
```
   
    
---
    
</b></i>     
</p>

---

In [27]:
dict_ = {x:round(mp.log(x, 2),3) for x in range(2,21,2)}
print(dict_)

{2: 1.0, 4: 2.0, 6: 2.585, 8: 3.0, 10: 3.322, 12: 3.585, 14: 3.807, 16: 4.0, 18: 4.17, 20: 4.322}


In [31]:
#b)

alfabeto ='ABCDEFGHIJKLMNOPQRSTUVWXYZ'
lista = ["C", "E","F", "I", "J", "L", "O", "R", "T"]

dict_ = {lista[i]:alfabeto.index(lista[i])+1  for i in range(0, len(lista))}
print(dict_)

#Outra forma de resolver
dicionario2 = {chr(64+num):num for num in range(1,21) if num%5==0 or num%3==0}
dicionario2

{'C': 3, 'E': 5, 'F': 6, 'I': 9, 'J': 10, 'L': 12, 'O': 15, 'R': 18, 'T': 20}


{'C': 3, 'E': 5, 'F': 6, 'I': 9, 'J': 10, 'L': 12, 'O': 15, 'R': 18, 'T': 20}

<hr style="border: 2px solid #4500ff;">

# <font color="#4500ff" style="font-size: 25px;">Exercício 116</font>
<hr style="border: 2px solid #4500ff;">

 <p style='font-size: 23px; line-height: 2; margin: 0px 0px; text-align: justify; text-indent: 0px;'>    
<i><b> a) Use compreensão de dicionários para extrair os produtos com preço superior a dois mil reais.

    
```python
produtos = {
    'geladeira':1619.1,
    'ps5':4999,
    'liquidificador':89.9,
    'notebook':3500,
    'TV':1500,
    'ventilador':79.99,
    'microondas':399.99
    }

   
```

### b) Considerando que cada valor deste dicionário representa o preço de cada produto, obtenha o custo total desta cesta de produtos.

---
    
</b></i>     
</p>

---

In [50]:
produtos = {
    'geladeira':1619.1,
    'ps5':4999,
    'liquidificador':89.9,
    'notebook':3500,
    'TV':1500,
    'ventilador':79.99,
    'microondas':399.99
    }

#a)
selecao = {chave:valor for chave,valor in produtos.items() if valor>2000}
print(selecao)


# 2º forma
dict_ = {x:produtos[x] for x in produtos if produtos[x] > 2000}
dict_

{'ps5': 4999, 'notebook': 3500}

In [56]:
# B)

soma = sum(list(produtos.values()))
soma

12187.98